net/addr.go

```diff
@@ -60,11 +60,7 @@ func (a Addr) String() string {
 
 // URL formats the scheme with address and with the optional port.
 func (a Addr) URL() string {
-	u := url.URL{
-		Scheme: a.scheme,
-		Host:   a.String(),
-	}
-	return u.String()
+	return a.scheme + "://" + a.String()
```

Comment:
Is this line changed from a standard `url.URL` because of the incorrect handling of IPc6 addresses? Or what it the reason behind?

Reply:
Yes. It escapes '%' so it returns '%25' instead of '%'.

Reply:
I see. Good catch!